File: Components/DevExpressVCL/Demos/ExpressSpreadSheet/CBuilder/HorizontalReport/HorizontalReportDemoMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "HorizontalReportDemoMain.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "BaseForm"
#pragma link "ReportPreviewUnit"
#pragma link "ReportDesignerBaseForm"
#pragma link "cxClasses"
#pragma link "cxLookAndFeels"
#pragma link "cxCheckBox"
#pragma link "cxContainer"
#pragma link "cxControls"
#pragma link "cxEdit"
#pragma link "cxGraphics"
#pragma link "cxGroupBox"
#pragma link "cxLabel"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxSplitter"
#pragma link "cxTextEdit"
#pragma link "dxSpreadSheetCore"
#pragma link "dxSpreadSheetReportDesigner"
#pragma link "cxFilterControl"
#pragma link "dxmdaset"
#pragma resource "*.dfm"
TfrmHorizontalReport *frmHorizontalReport;
//---------------------------------------------------------------------------
__fastcall TfrmHorizontalReport::TfrmHorizontalReport(TComponent* Owner)
	: TfrmReportDesignerBase(Owner)
{
}
//---------------------------------------------------------------------------


void TfrmHorizontalReport::Initialize()
{
  LoadDataset(mdsEmployees, "..\\..\\Data\\Employees.mds");
  ReportDesigner->LoadFromFile("..\\..\\Data\\HorizontalReportTemplate.xlsx");
  ReportDesigner->Options->Orientation = roHorizontal;
  TfrmReportDesignerBase::Initialize();
}

File: Components/DevExpressVCL/Demos/ExpressSpreadSheet/Delphi/HorizontalReport/HorizontalReportDemoMain.pas
unit HorizontalReportDemoMain;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms, Dialogs, ReportDesignerBaseForm, cxGraphics, cxControls, cxLookAndFeels,
  cxLookAndFeelPainters, cxContainer, cxEdit, DB, dxmdaset, cxClasses, Menus, StdCtrls, cxFilterControl, cxCheckBox, cxGroupBox, ExtCtrls, cxSplitter, 
  dxSpreadSheetCore, dxSpreadSheetReportDesigner, ComCtrls;

type
  TfrmHorizontalReport = class(TfrmReportDesignerBase)
    mdsEmployees: TdxMemData;
    dsEmployees: TDataSource;
  private
    { Private declarations }
  public
    procedure Initialize; override;
  end;

var
  frmHorizontalReport: TfrmHorizontalReport;

implementation

{$R *.dfm}

procedure TfrmHorizontalReport.Initialize;
begin
  LoadDataset(mdsEmployees, '..\..\Data\Employees.mds');
  ReportDesigner.LoadFromFile('..\..\Data\HorizontalReportTemplate.xlsx');
  ReportDesigner.Options.Orientation := roHorizontal;
  inherited Initialize;
end;

end.
